src/pages/index.js

```diff
@@ -5,52 +5,348 @@ import PopupWithForm from "../components/PopupWithForm.js";
+        avatarChangeModal.close();
+      })
+      .catch((err) => console.error("Error updating avatar:", err))
+      .finally(() => {
+        avatarSubmitButton.textContent = "Save";
+      });
+  } else {
+    console.error("Avatar URL is not defined");
+    avatarSubmitButton.textContent = "Save"; // Ensure the button text is reset
+  }
+}
+
+const avatarFormElement = document.querySelector("#avatar-form");
+if (avatarFormElement) {
+  avatarFormElement.addEventListener("submit", (event) => {
+    event.preventDefault();
+    const inputValues = new FormData(event.target);
+    handleAvatarChangeSubmit(Object.fromEntries(inputValues));
+  });
+}
+
+// Event Listener for opening avatar modal
+avatarImage.addEventListener("click", () => {
+  submitButtonSelector.textContent = "Save";
+  avatarChangeModal.open();
+  avatarFormValidator.toggleButtonState();
+});
+console.log("User object:", user);
+
+// ----------------------------
+
+// Delete card modal
+
+const deleteModal = document.querySelector("#delete-modal");
+const deleteModalCloseButton = document.querySelector("#delete-close-modal");
+const deleteConfirmForm = document.querySelector("#delete-confirm-form");
+const deleteConfirmModal = new ConfirmPopup(
+  "#delete-modal",
+  handleCardDeleteSubmit
+);
+deleteConfirmModal.setEventListeners();
+
+let cardToDelete = null;
+
+// Add event listeners to open the modal when delete button is clicked
+document.querySelectorAll(".card__delete-button").forEach((button) => {
+  button.addEventListener("click", (event) => {
+    event.preventDefault();
+    deleteModal.style.display = "block";
+    cardToDelete = event.target.closest(".card");
+  });
+});
+
+// Add event listener to close the modal
+deleteModalCloseButton.addEventListener("click", () => {
+  deleteModal.style.display = "none";
+  cardToDelete = null;
+});
+
+deleteConfirmForm.addEventListener("submit", (event) => {
+  event.preventDefault();
+  if (cardToDelete) {
+    // const cardId = cardToDelete._data.id;
+
+    handleDeleteClick(cardToDelete);
+    deleteModal.style.display = "none";
+    cardToDelete = null;
+  }
+});
+
+function handleDeleteClick(card) {
+  deleteConfirmModal.open(card);
+  cardToDelete = card;
+}
+
+const deleteSubmitButton = deleteConfirmForm.querySelector(
+  "button[type='submit']"
+);
+
+function handleCardDeleteSubmit(card) {
+  deleteSubmitButton.textContent = "Saving...";
+  const card_id = card._id;
+  console.log("Card ID:", card_id); // Debugging log
+
+  api
+    .deleteCard(card_id)
+    .then((message) => {
+      console.log(message);
+      deleteConfirmModal.close();
+      card.handleDeleteCard(); // Assuming this method exists to remove the card from the DOM
+    })
+    .catch((error) => {
+      console.error("Error deleting card:", error);
+    })
+    .finally(() => {
+      deleteSubmitButton.textContent = "Yes"; // Reset button text after completion
+    });
+}
+
+deleteConfirmForm.addEventListener("submit", (event) => {
```

Comment:
This modal is handled by `ConfirmPopup`.  


And the form should be handled also there

Reply:
so i should remove this from index.js?

Reply:
> so i should remove this from index.js?

yes